libPhoneNumber/NBPhoneNumberOfflineGeocoder.m

```diff
@@ -0,0 +1,173 @@
+//
+//  NBPhoneNumberOfflineGeocoder.m
+//  libPhoneNumberiOS
+//
+//  Created by Rastaar Haghi on 6/12/20.
+//  Copyright © 2020 ohtalk.me. All rights reserved.
+//
+
+#import "NBPhoneNumberOfflineGeocoder.h"
+
+@implementation NBPhoneNumberOfflineGeocoder {
+ @private
+  NBPhoneNumberUtil *phoneUtil;
+  NBGeocoderMetadataHelper *geocoderHelper;
+}
+
+- (instancetype)init {
+  self = [super init];
+  self->phoneUtil = NBPhoneNumberUtil.sharedInstance;
+  // gather all available language database files
+  NSBundle *bundle = [NSBundle bundleForClass:self.classForCoder];
+  NSURL *bundleURL = [[bundle resourceURL] URLByAppendingPathComponent:@"Resources.bundle"];
+  NSArray *filesInResourceBundle =
+      [[NSFileManager defaultManager] contentsOfDirectoryAtURL:bundleURL
+                                    includingPropertiesForKeys:NULL
+                                                       options:0
+                                                         error:NULL];
+  NSMutableArray *supportedLanguages = [[NSMutableArray alloc] init];
+  [filesInResourceBundle enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
+    NSString *filename = (NSString *)obj;
+    if (filename != NULL) {
+      [supportedLanguages addObject:[[filename lastPathComponent] stringByDeletingPathExtension]];
+    }
+  }];
+  NSString *language;
+  // need to get from ios device settings
+  if ([[NSLocale preferredLanguages] count] > 0) {
+    // set language to preferred
+    language = [NSLocale preferredLanguages][0];
+    // if the preferred language isn't supported by metadata, set by default to english
+    if (![supportedLanguages containsObject:language]) {
+      language = @"en";
+      NSLog(@"Set default language to ENGLISH");
+    }
+  } else {
+    language = @"en";
+  }
+
+  self->geocoderHelper = [[NBGeocoderMetadataHelper alloc] initWithCountryCode:@1
+                                                                  withLanguage:language];
+  return self;
+}
+
+- (NSString *)countryNameForNumber:(NBPhoneNumber *)number
+                  withLanguageCode:(NSString *)languageCode {
+  if (number == NULL) return @"";
+  NSArray *regionCodes = [self->phoneUtil getRegionCodesForCountryCode:number.countryCode];
+  if ([regionCodes count] == 1) {
+    return [self regionDisplayName:[regionCodes objectAtIndex:0] withLanguageCode:languageCode];
```

Comment:
nit: Instead of `[regionCodes objectAtIndex:0]`, you can use `[regionCodes firstObject]`

Reply:
regionCodes[0]

Reply:
done.